### PR TITLE
#169383181 Improve data flow

### PR DESCRIPTION
**Pull request by NiyoEric**
#### What does this PR do?
Improve a network request so that we can call API once.

#### Description of Task to be completed?

**Expected**
- API request should be called once at a time
- Create *helpers* and *images* directories
- Use JS DOC to explain code

#### How should this be manually tested?
- `git clone ch-improve-data-flow-169383181` branch
- cd OddWeather
- `git checkout ch-improve-data-flow-169383181`
- `npm install`
- create .env file 
- `npm start`

#### What are the relevant pivotal tracker stories?
[#169383181](https://www.pivotaltracker.com/story/show/169383181)

#### Screenshots (if appropriate)
- Network request
Screen Shot 2019-10-29 at 07 15 23

- Performance
Screen Shot 2019-10-29 at 07 29 45

#### Questions:
- Check if port is not blocked before
- make sure to have .env file

### CODE REVIEW

**Review by hezzie — APPROVED** [review]

**Review by victorkarangwa — APPROVED** [review]

File: src/components/Title.js [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
         return (
             <div>
                 <h6>Odd Weather</h6>
                 {!error && 
-                    <div class="weatherGrid">
-                        <div class="degree-span-row-2"><h2>{convertDegrees(data.list === undefined ? "": data.list[0].main.temp)}&#176;C</h2></div>
+                    <div className="weatherGrid">
+                        <div className="degree-span-row-2"><h2>{convertDegrees(data.list === undefined ? "": data.list[0].main.temp)}&#176;C</h2></div>
                         <div><p id="city">{data.list === undefined ? "" : data.city.name}</p></div>
                         <div id="icon"><img alt="" src = { `http://openweathermap.org/img/w/${data.list === undefined ? "" : data.list[0].weather[0].icon}.png`}/></div>
```
Comment (kiproping) [inline src/components/Title.js]: Remove URL's

File: src/helpers/currentLocation.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Get a city or coordinate from a state.
+ * @returns {city} forecast given city.
+ */
+const currentLocation = ({city, cord}) => {
```
Comment (kiproping) [inline src/helpers/currentLocation.js]: You can put all these functions in one file.

File: src/App.js [hunk 3]
```diff
@@ -21,13 +23,10 @@ class App extends Component {
...
     };
   }
 
-  options = {
-    enableHighAccuracy: true,
-    timeout: 5000
-  };
   sleep = ms => {
     return new Promise(resolve => setInterval(resolve, ms));
   };
```
Comment (kiproping) [inline src/App.js]: As we discussed, no sleep or setInterval.

Comment (kipropbrian) [inline src/App.js:32]: As we have discussed no setInterval

File: src/components/Title.js (hunk 1 above)
Comment (kipropbrian) [inline src/components/Title.js:15]: Remove the URL

File: src/App.js (hunk 3 above)
Comment (NiyoEric) [reply src/App.js:32]: @victorkarangwa is working on it?

File: src/components/Title.js (hunk 1 above)
Comment (NiyoEric) [reply src/components/Title.js:15]: removed

**Review by WillyBoris — APPROVED** [review]

**Pull request merged**